src/database/Database.h

```diff
@@ -191,6 +209,11 @@ class Database : NonMovableOrCopyable
     // Check schema version and apply any upgrades if necessary.
     void upgradeToCurrentSchema();
 
+    // The name of the column containing the ledger entry extension,
+    // which appears in all the tables which contain any type of
+    // ledger entry.
+    static std::string const ledgerExtName;
```

Comment:
> Does this really add clarity though? For example, to remember the name of the constant you must remember "ledgerExt" _plus_ an additional word, whereas the constant is just "ledgerext". After reviewing more of the PR, I'm becoming more convinced this is not adding clarity.

Which additional word?  Do you mean ```Database::```?

I agree that it does not read particularly well, but I did that because I wanted to have the compiler ensure first that the LedgerEntry extension had the same name in each table in which it appears (that part _is_ just for clarity) and that the compiler would prevent any misspellings.  But it does make the client code more verbose and less readable.  I'll change it if you think it's a better tradeoff the other way.  (If you think the compiler-time consistency is worth the client-code verbosity, then I'll just change the constant name to match our convention.)

What I'd prefer to both options would be for there to be a separate module that factors out all the raw SQL generation and allowed clients to write all their uses from methods such as ```generateGenericSelectSQL()``` and ```generatePostgresUpdateSQL()```, which would protect from both misspellings and argument mismatches (in particular, all those client wrapper methods could be unit-tested -- the missing "v12" that you caught below in code review only turned up in practice when the buckets were >70% through being replayed, so some strings are quite rarely used!) and make the client code easier rather than harder to read (standard C++ method calls rather than long strings of not-quite-repeated raw SQL throughout all the *SQL.cpp files).  But that'd be a PR of its own, not something to clutter this one with.

Reply:
I mean remembering the name "ledgerExtName" means you have already remembered "ledgerext", so it doesn't save any mental effort at all. I don't think the compile-time consistency helps much here: if there were a mis-spelling then the code definitely will not pass our acceptance tests.

I agree that factoring out raw SQL generation sounds like a really good idea. Almost all of our queries have a few standard parts (`WITH` statements for example) that we can use as components. But as you note, "that'd be a PR of its own." How about you open an issue to address the bigger problem, and stick with hard coded here?

Reply:
> How about you open an issue to address the bigger problem, and stick with hard coded here?

That sounds good to me.  I've changed to code to use a hardcoded "ledgerext", as with other columns, and have filed issue #2594 .